### ISSUE TITLE
WooCommerce: Store credit cannot be used

### ISSUE THREAD

**Issue by michaelbeil** [AffiliateWP/affiliatewp-store-credit#108]
## Bug Report
When you are at checkout and you have a store credit balance that you can use for the order, a notice appears: **"You have an account balance of XX.XX. Would you like to use it now?"** You can apply the store credit and the price goes down, but when you try to place the order, it says this: `"Coupon affiliate-credit-202002141427_1 was used in another transaction during this checkout, and coupon usage limit is reached. Please remove the coupon and try again." 
` 

I was able to replicate the problem:
![Screen Capture on 2020-02-14 at 21-38-50](https://user-images.githubusercontent.com/4752949/74565937-969da000-4f72-11ea-8d9d-a5ce2b9b54e6.gif)

### Steps to reproduce the behavior 
1) apply store credit at checkout
2) place the order and you will see the error 

### Information 
PHP Version: 7.3.12
WordPress Version: 5.3.2
WooCommerce Version: 3.9.1
AffiliateWP Version: 2.4.3
Store Credit Version: 2.3.2

Any other relevant information:  
First reported on this [ticket](https://secure.helpscout.net/conversation/1073432586/144554?folderId=634609).

PR: #109

**Comment by gspice** [comment]
I reproduced this issue as well on my local Woo test site.  It seems there was a core WooCommerce update in `3.9` where native-generated coupons still work fine, but any outside plugin that creates a WooCommerce Coupon is failing until there are code changes made.    This thread on .org for a Woo coupon plugin talks more about it, and hopefully can shed some light on what we have to do to fix it.   

https://wordpress.org/support/topic/auto-generated-coupon-code-not-working/

Bottom line is for our customers that use Store Credit, after they update to Woo `3.9` it will no longer work .

**Comment by michaelbeil** [comment]
A [customer](https://secure.helpscout.net/conversation/1081781956/145364?folderId=205529) made an update to WooCommerce 3.9.2, and now store credit does not work:

>It did work and then stopped working.

I am letting them know that we will address this in the future.

**Comment by alexstandiford** [comment]
I have identified the source of this problem, thanks to @gspice's for the link - it pointed me in the right direction!

I'm going to submit a PR to resolve this permanently, but for the short-term this can be added to a customer's functions.php file to resolve:

```php
add_filter(  'affwp_store_credit_woocommerce_coupon_data', function( $data ){
    $data['usage_count'] = '0';
    
    return $data;
} );
```

This issue happens because Woo expects a specific piece of metadata to be included in generated coupons. Without this data, the coupon does not work because it assumes that the coupon has already been used. The above filter hooks into store credit and adds this metadata.

**Comment by alexstandiford** [comment]
PR #109

**Comment by michaelbeil** [comment]
The PR fixes the problem: 
![Screen Capture on 2020-02-24 at 17-37-04](https://user-images.githubusercontent.com/4752949/75172335-618d0c80-572d-11ea-8c22-238e672aa31f.gif)
